ql/pricingengines/swap/discretizedswap.cpp

```diff
@@ -19,8 +19,17 @@
 */
 
 #include <ql/pricingengines/swap/discretizedswap.hpp>
+#include <ql/settings.hpp>
 
 namespace QuantLib {
+    namespace {
+        inline bool useCouponInPostAdjust(const Time& resetTime,
+                                          const Time& payTime,
+                                          const boost::optional<bool>& includeTodaysCashFlows) {
+            return (resetTime < 0.0) &&
+                   ((payTime > 0.0) || (includeTodaysCashFlows && (payTime == 0.0)));
```

Comment:
I'd try to measure if there's a difference in time between passing `includeTodaysCashFlows` and calling `Settings::instance().includeTodaysCashFlows()` here instead (so it can be accessed only if `payTime == 0`).  We could also do it in the `DiscretizedSwap` constructor, since a new one is created for each calculation.